src/app/qgshandlebadlayers.h

```diff
@@ -63,18 +63,35 @@ class APP_EXPORT QgsHandleBadLayers
     void editAuthCfg();
     void apply();
     void accept() override;
+    void autoFind();
 
   private:
     QPushButton *mBrowseButton = nullptr;
     QPushButton *mApplyButton = nullptr;
+    QPushButton *mAutoFindButton = nullptr;
     const QList<QDomNode> &mLayers;
     QList<int> mRows;
     QString mVectorFileFilter;
     QString mRasterFileFilter;
-    QHash <QString, QList<QString> > mFileBase;
+    // Registry of the original paths associated with a file as a backup
+    QHash <QString, QString > mOriginalFileBase;
+    // Keeps a registry of valid alternatives for a basepath
+    QHash <QString, QList<QString> > mAlternativeBasepaths;
 
     QString filename( int row );
     void setFilename( int row, const QString &filename );
+    /*checkBasepath will check if the newPath for the provided name is valid.
```

Comment:
```suggestion
    /**
     * Checks if \a newPath for the provided \a name is valid.
```

It doesn't really mather a lot since it's private and in app. But if it's ever moved, it's nice if it's already properly formatted :)

Reply:
Thanks